src/monster.h

```diff
@@ -129,13 +129,21 @@ class Monster final : public Creature
 	BlockType_t blockHit(Creature* attacker, CombatType_t combatType, int32_t& damage, bool checkDefense = false,
 	                     bool checkArmor = false, bool field = false, bool ignoreResistances = false) override;
 
+	// map cache
+	int32_t getWalkCache(const Position& pos) const;
+
 	// monster icons
 	MonsterIconHashMap& getSpecialIcons() { return monsterIcons; }
 	const MonsterIconHashMap& getSpecialIcons() const { return monsterIcons; }
 
 	static uint32_t monsterAutoID;
 
 private:
+	static constexpr int32_t walkCacheWidth = Map::maxViewportX * 2 + 1;
+	static constexpr int32_t walkCacheHeight = Map::maxViewportY * 2 + 1;
+	static constexpr int32_t maxWalkCacheWidth = (walkCacheWidth - 1) / 2;
+	static constexpr int32_t maxWalkCacheHeight = (walkCacheHeight - 1) / 2;
```

Comment:
Aren't these just

```suggestion
	static constexpr int32_t maxWalkCacheWidth = Map::maxViewportX;
	static constexpr int32_t maxWalkCacheHeight = Map::maxViewportY;
```
?

It's just reversing the calculation

Reply:
Used map constexpr direcly